packages/react-devtools-shared/src/backend/console.js

```diff
@@ -7,21 +7,27 @@
  * @flow
  */
 
-import {getInternalReactConstants} from './renderer';
-import describeComponentFrame from './describeComponentFrame';
-
 import type {Fiber} from 'react-reconciler/src/ReactInternalTypes';
-import type {ReactRenderer} from './types';
+import type {CurrentDispatcherRef, ReactRenderer, WorkTagMap} from './types';
+
+import {getInternalReactConstants} from './renderer';
+import {getStackByFiberInDevAndProd} from './DevToolsFiberComponentStack';
 
 const APPEND_STACK_TO_METHODS = ['error', 'trace', 'warn'];
 
-const FRAME_REGEX = /\n {4}in /;
+// React's custom built component stack strings match "\s{4}in"
+// Chrome's prefix matches "\s{4}at"
+const PREFIX_REGEX = /\s{4}(in|at)\s{1}/;
+// Firefox and Safari have no prefix ("")
+// but we can fallback to looking for location info (e.g. "foo.js:12:345")
+const ROW_COLUMN_NUMBER_REGEX = /:\d+:\d+$/;
```

Comment:
Hm. This will only match if the very last line has a line number but that's not always true since we sometimes bail out. It's also possible that the root is a built-in component. E.g. `<Suspense>`. Those won't have a line number. It's probably likely that at least one of them as one but not guaranteed.

Reply:
Right, this is not a great check but it's the fallback check. Do you have a better suggestion?

I can't just search for a word like "Suspense" obviously :wink:

Reply:
> It's also possible that the root is a built-in component. E.g. <Suspense>. Those won't have a line number. It's probably likely that at least one of the frames has one but not guaranteed.

Since the component stack is reverse, the "first" item would be a user component anyway- but this should match an entry on a line other than the first regardless.

I'm banking on there being at least one line with source info, or we won't have a way to differentiate between a stack and an arbitrary string.

Reply:
`/:\d+:\d+(\n|$)/` would at least catch if there's any numeric patterns for any of the rows. That might be enough.

Reply:
Ah right. I forgot the `m` flag to make `/:\d+:\d+$/` function that way.